### PR TITLE
Root interface changes

### LINKED ISSUE #426: ROOT TGeoMatrix interface change
Dear DD4Hep developers,

dd4hep version 01-05 (and, I think, master as well, since the code in question is unchanged) fails to build with gcc 6.2 (or newer) and HEAD version of ROOT. The error message is "invalid initialization of non-const reference of type ‘TGeoMatrix&’ from an rvalue of type ‘TGeoMatrix’". 

Thanks,
Ivan for the GENSER team

 - OS version: SLC6 (not tested, but will likely fail with Centos 7)
 - Compiler version: GCC 6.2, 7.3.0, 8.1.0
 - DD4hep version: tag v01-05
 - Reproduced by: building against ROOT-head.
 - Build log: https://cernbox.cern.ch/index.php/s/TxSdekx7XSLl9bA
Important part:
```
[ 44%] Building CXX object UtilityApps/CMakeFiles/geoConverter.dir/src/converter.cpp.o
/mnt/build/jenkins/workspace/lcg_personal_experimental/BUILDTYPE/Release/COMPILER/gcc62binutils/LABEL/slc6/build/projects/DD4hep-01-05/src/DD4hep/01-05/DDCore/src/plugins/LCDDConverter.cpp: In member function ‘virtual xml_h dd4hep::detail::LCDDConverter::handleSolid(const string&, const TGeoShape*) const’:
/mnt/build/jenkins/workspace/lcg_personal_experimental/BUILDTYPE/Release/COMPILER/gcc62binutils/LABEL/slc6/build/projects/DD4hep-01-05/src/DD4hep/01-05/DDCore/src/plugins/LCDDConverter.cpp:552:39: error: invalid initialization of non-const reference of type ‘TGeoMatrix&’ from an rvalue of type ‘TGeoMatrix’
         TGeoMatrix& linv = lm->Inverse()
                            ~~~~~~~~~~~^~
/mnt/build/jenkins/workspace/lcg_personal_experimental/BUILDTYPE/Release/COMPILER/gcc62binutils/LABEL/slc6/build/projects/DD4hep-01-05/src/DD4hep/01-05/DDCore/src/plugins/LCDDConverter.cpp:571:39: error: invalid initialization of non-const reference of type ‘TGeoMatrix&’ from an rvalue of type ‘TGeoMatrix’
         TGeoMatrix& rinv = rm->Inverse()
                            ~~~~~~~~~~~^~
[ 44%] Building CXX object DDG4/CMakeFiles/DDG4.dir/src/Geant4Exec.cpp.o
[ 44%] Building CXX object DDG4/CMakeFiles/DDG4.dir/src/Geant4Field.cpp.o
[ 44%] Building CXX object DDCond/CMakeFiles/DDCond.dir/src/ConditionsOperators.cpp.o
[ 44%] Building CXX object DDEve/CMakeFiles/DDEve.dir/src/DDEvePlugins.cpp.o
make[6]: *** [DDCore/CMakeFiles/DDCorePlugins.dir/src/plugins/LCDDConverter.cpp.o] Error 1
```

### PR DESCRIPTION
**Pull request by andresailer**
BEGINRELEASENOTES
- Adapt to root interfaces changes for `TGeoMatrix::Inverse` (https://github.com/root-project/root/pull/2365), fixes #426 

ENDRELEASENOTES

### CODE REVIEW

**Comment by ianna** [comment]
@mrodozov - FYI

File: DDCore/src/MatrixHelpers.cpp [hunk 1]
```diff
@@ -112,6 +112,19 @@ Transform3D dd4hep::detail::matrix::_transform(const TGeoMatrix* matrix)    {
                      0e0,0e0,0e0,t[1]*MM_2_CM,
                      0e0,0e0,0e0,t[2]*MM_2_CM);
 }
+// add another implementation that takes const reference
+Transform3D dd4hep::detail::matrix::_transform(const TGeoMatrix& matrix)    {
+  const Double_t* t = matrix.GetTranslation();
+  if ( matrix.IsRotation() )  {
+    const Double_t* r = matrix.GetRotationMatrix();
+    return Transform3D(r[0],r[1],r[2],t[0]*MM_2_CM,
```
Comment (ianna) [inline DDCore/src/MatrixHelpers.cpp]: There are a lot of conversions here. It looks like the MM_2_CM and RAD_2_DEGREE assume that the input units are mm and rad. Should it be a user who defines which units to use and if they are not the DD4hep default, convert them?

**Comment by MarkusFrankATcernch** [comment]
Updates are fine. We requested them in ROOT, 
so we should be prepared to update our code accordingly.

File: DDCore/src/MatrixHelpers.cpp (hunk 1 above)
Comment (petricm) [reply DDCore/src/MatrixHelpers.cpp]: The user should always use DD4hep units, when addressing/talking to DD4hep. There some exceptions to this rule, e.g. when interfaces of underlying classes are exposed, there you should use or expect ROOT or Geant4 conventions.

**Pull request merged**